File: Task3/Javascript/inoutdiamond.js
const fs = require('fs');

fs.readFile('input.txt', 'utf8', (err, data) => {
    if (err) {
        console.error('Error reading file:', err);
        return;
    }

    const n = parseInt(data.trim(), 10);
    if (isNaN(n) || n <= 0) {
        console.error('Invalid number in input file.');
        return;
    }

    let diamond = '';

    for (let i = 1; i <= n; i++) {
        diamond += ' '.repeat(n - i) + '*'.repeat(2 * i - 1) + '\n';
    }

    for (let i = n - 1; i >= 1; i--) {
        diamond += ' '.repeat(n - i) + '*'.repeat(2 * i - 1) + '\n';
    }

    fs.writeFile('output.txt', diamond, err => {
        if (err) {
            console.error('Error writing file:', err);
        }
    });
});

File: Task3/Go/inoutdiamond.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func main() {
	inputFile := "input.txt"
	outputFile := "output.txt"

	content, err := ioutil.ReadFile(inputFile)
	if err != nil {
		fmt.Println("Error reading file:", err)
		return
	}

	n, err := strconv.Atoi(strings.TrimSpace(string(content)))
	if err != nil || n <= 0 {
		fmt.Println("Invalid number in input file.")
		return
	}

	var result strings.Builder

	for i := 1; i <= n; i++ {
		result.WriteString(strings.Repeat(" ", n-i))
		result.WriteString(strings.Repeat("*", 2*i-1))
		result.WriteString("\n")
	}

	for i := n - 1; i >= 1; i-- {
		result.WriteString(strings.Repeat(" ", n-i))
		result.WriteString(strings.Repeat("*", 2*i-1))
		result.WriteString("\n")
	}

	err = ioutil.WriteFile(outputFile, []byte(result.String()), 0644)
	if err != nil {
		fmt.Println("Error writing to file:", err)
	}
}

